roles/download/tasks/main.yml

```diff
@@ -26,5 +59,29 @@
     - not skip_downloads | default(false)
     - download.enabled
     - item.value.enabled
-    - (not (item.value.container | default(false))) or (item.value.container and download_container)
+    - >
+      (
+        not (
+          item.value.container | default(false)
+        )
+      )
+      or (
+        item.value.container and
+        download_container and
```

Comment:
You should be able to get rid of those conditions by filtering the list before feeding it to the loop instead:
```yaml 
vars:
    base_dls: "{{ downloads | combine(kubeadm_images) | dict2items }}"
    containers_dls: "{{ base_dls | selectattr('value.container', 'defined') | selectattr('value.container') }}"